pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -88,7 +97,58 @@ func (client *richOvirtClient) GetVM(id *string, name *string, cluster *string)
 	return vm, nil
 }
 
-func (client *richOvirtClient) fetchVM(id *string, name *string, cluster *string) (*ovirtsdk.Vm, error) {
+// RenameVM rename the vm
+func (client *richOvirtClient) RenameVM(id string, newName string) error {
+	_, err := client.connection.SystemService().VmsService().VmService(id).
+		Update().
+		Vm(
+			ovirtsdk.NewVmBuilder().Name(newName).MustBuild(),
+		).
+		Send()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// StopVM stop the VM and wait for the vm to be stopped
+func (client *richOvirtClient) StopVM(id string) error {
+	vmService := client.connection.SystemService().VmsService().VmService(id)
+
+	// Return if VM is already stopped:
+	tmp, _ := vmService.Get().Send()
+	if status, _ := tmp.MustVm().Status(); status == ovirtsdk.VMSTATUS_DOWN {
+		return nil
+	}
+
+	// Stop the VM gracefully:
+	_, err := vmService.Shutdown().Send()
+	if err != nil {
+		return err
+	}
+
+	// Wait for VM to be stopped
+	c := make(chan bool, 1)
+	go func() {
+		for {
+			time.Sleep(vmPollInterval * time.Second)
+			tmp, _ = vmService.Get().Send()
+			if status, _ := tmp.MustVm().Status(); status == ovirtsdk.VMSTATUS_DOWN {
```

Comment:
Is it possible the VM disappears (i.e. someone deletes it on the oVirt side)? If yes, I think that we can panic here.

Reply:
Right, thanks. I will fix. I'll get rid of all 'Must' methods, I used them for testing only, and forgot to remove them all.